recipe/windows/deactivate.bat

```diff
@@ -0,0 +1,41 @@
+:: Restore environment variables (if there is anything to restore)
+if not "%CUDA_HOME_CONDA_NVCC_BACKUP%"=="" (
+    set "CUDA_HOME=%CUDA_HOME_CONDA_NVCC_BACKUP%"
+    set "CUDA_HOME_CONDA_NVCC_BACKUP="
+)
+
+if not "%CUDA_PATH_CONDA_NVCC_BACKUP%"=="" (
+    set "CUDA_PATH=%CUDA_PATH_CONDA_NVCC_BACKUP%"
+    set "CUDA_PATH_CONDA_NVCC_BACKUP="
+    :: If there's a backup for CUDA_PATH but not for CUDA_HOME
+    :: that means we aliased it during env activation
+    if "%CUDA_HOME_CONDA_NVCC_BACKUP%"=="" (
+        set "CUDA_HOME="
+    )
+)
+
+if not "%CFLAGS_CONDA_NVCC_BACKUP%"=="" (
+    set "CFLAGS=%CFLAGS_CONDA_NVCC_BACKUP%"
+    set "CFLAGS_CONDA_NVCC_BACKUP="
+)
+
+if not "%CPPFLAGS_CONDA_NVCC_BACKUP%"=="" (
+    set "CPPFLAGS=%CPPFLAGS_CONDA_NVCC_BACKUP%"
+    set "CPPFLAGS_CONDA_NVCC_BACKUP="
+)
+
+if not "%CXXFLAGS_CONDA_NVCC_BACKUP%"=="" (
+    set "CXXFLAGS=%CXXFLAGS_CONDA_NVCC_BACKUP%"
+    set "CXXFLAGS_CONDA_NVCC_BACKUP="
+)
+
+:: JRG: Should CONDA_BUILD_SYSROOT be replaced with LIBRARY_PREFIX?
+:: set "CONDA_BUILD_SYSROOT=%CONDA_PREFIX%"
+
+:: Remove or restore `cuda.lib` from the compiler sysroot.
+set "LIBCUDA_SO_CONDA_NVCC_BACKUP=%LIBRARY_LIB%\cuda.lib-conda-nvcc-backup"
+if exist "%LIBCUDA_SO_CONDA_NVCC_BACKUP%" (
+    ren "%LIBCUDA_SO_CONDA_NVCC_BACKUP%" "%LIBRARY_LIB%\cuda.lib"
+) else (
+    del "%LIBRARY_LIB%\cuda.lib"
```

Comment:
Can this fail somehow? If it does and the build passes, we'll be packaging this against its license.

Reply:
I can add an `errorlevel` check there and also text for non-existence in the `meta.yaml` tests. Would that be enough?

Reply:
See now, together with https://github.com/conda-forge/nvcc-feedstock/pull/49/files#diff-e3a5306c0a02b6cefeced655643e5e6a4625e9c8af54650c829f7931ca44009dR74-R80

Reply:
But that test is only for this recipe. What happens in downstream packages?

How about generating this `cuda.lib` ourselves from the DLL so that we can redistribute it?

Reply:
Something [like this](https://stackoverflow.com/questions/9360280/how-to-make-a-lib-file-when-have-a-dll-file-and-a-header-file)?

These are the DLLs available under `\bin`. Which one applies here?

```
Directory: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0\bin


    Length Name
    ------ ----
  89057280 cublas64_11.dll
 167523328 cublasLt64_11.dll
    335872 cudart32_110.dll
    401408 cudart64_110.dll
 162878976 cufft64_10.dll
    272896 cufftw64_10.dll
   1305600 cuinj64_110.dll
  52446720 curand64_10.dll
 525173248 cusolver64_10.dll
 314601984 cusolverMg64_10.dll
 159789568 cusparse64_11.dll
    232448 nppc64_11.dll
  10907648 nppial64_11.dll
   4569600 nppicc64_11.dll
   7840256 nppidei64_11.dll
  51866112 nppif64_11.dll
  26065920 nppig64_11.dll
   6411264 nppim64_11.dll
  20231168 nppist64_11.dll
    203264 nppisu64_11.dll
   2856960 nppitc64_11.dll
   8873472 npps64_11.dll
    299520 nvblas64_11.dll
   3133952 nvjpeg64_11.dll
   5118976 nvrtc-builtins64_110.dll
  16161792 nvrtc64_110_0.dll
```

Reply:
> Something like this?

Yes

> These are the DLLs available under \bin. Which one applies here?

Have a look at `cuda.lib` (It's a text file) It should tell you

Reply:
Thanks! Had to use `dumpbin /ALL cuda.lib` to figure it out. FIle is `nvcuda.dll`, which should be part of the drivers at `C:\Windows\system32`, but I can't find it in my system. [A separate installation process for the drivers](https://forums.developer.nvidia.com/t/cuda-rc-11-0-installs-a-nvcuda-dll-that-does-not-have-version-information/127865) seems to fix it.

Reply:
I added a stub now. Tried to generate off the drivers installation in the CI, but `nvcuda.dll` is not present there (I guess because no GPU). So I generated one from my 456.71 Windows installation and copied it over. Is that ok? I left a `generate_stub.bat` script in case it needs to be regenerated again in the future (drivers update?).

Reply:
the file might get outdated in that case. How about this? create an empty `cuda.lib` in `bld.bat` here so that the empty file will get packaged and then overriding that in a `activate.bat`? This way, conda will not package the actual `cuda.lib` in a downstream package because that file is already there.

Reply:
Oh, that could work, yes. Check last commit 36ade89.